Source/am_map.c

```diff
@@ -742,90 +724,77 @@ boolean AM_Responder
   int rc;
   static int bigstate=0;
   static char buffer[20];
-  int ch;                                                       // phares
-  static int joywait = 0;
 
   rc = false;
 
-  // [FG] automap joystick button
-  if (ev->type == ev_joystick && joywait < I_GetTime())
-  {
-    // [FG] crude hack converting the joystick button press into a key press
-    if (joybautomap > -1 && (ev->data1 & (1 << joybautomap)))
-    {
-      ev->type = ev_keydown;
-      ev->data1 = key_map;
-      joywait = I_GetTime() + 5;
-    }
-  }
-
   if (!automapactive)
   {
-    if (ev->type == ev_keydown && ev->data1 == key_map)         // phares
+    if (M_InputActivated(input_map))
     {
       AM_Start ();
       viewactive = false;
       rc = true;
     }
   }
-  // [crispy] zoom automap with the mouse (wheel)
-  else if (ev->type == ev_mouse && !menuactive)
-  {
-    if (mousebprevweapon >= 0 && ev->data1 & (1 << mousebprevweapon))
-    {
-      mtof_zoommul = M2_ZOOMOUT;
-      ftom_zoommul = M2_ZOOMIN;
-      rc = true;
-    }
-    else
-    if (mousebnextweapon >= 0 && ev->data1 & (1 << mousebnextweapon))
-    {
-      mtof_zoommul = M2_ZOOMIN;
-      ftom_zoommul = M2_ZOOMOUT;
-      rc = true;
-    }
-  }
-  else if (ev->type == ev_keydown)
+  else if (ev->type == ev_keydown ||
```

Comment:
Could this get abstracted (just a thought)?

Reply:
I think not worth it. We can remove these checks, I kept them to be closer to the original code.

Reply:
OK.